### ISSUE TITLE
Implement fish support

### ISSUE THREAD

**Issue by DarthGandalf** [kvirc/KVIrc#318]
**Reported by irc://irc.freenode.org/Meliorator on 14 Dec 2008 13:21:49 UTC**
It just seems to occur with key lengths that are > 56 char's and/or have patterns of certain char's. When I first experienced the issue (a long time ago) I discussed it with the mircryption dev' called mouser (on EFnet) and he suspected that key lengths and certain char's were not handled correctly. I had a key that I knew broke on KVIrc, which I've lost now.

---

Migrated from: https://svn.kvirc.de/kvirc/ticket/318

**Comment by DarthGandalf** [comment]
**Comment by irc://irc.freenode.org/Meliorator on 14 Dec 2008 13:33:19**
This is generally the accpeted form of mircryption/blowfish implementation: http://fish.sekure.us/

**Comment by DarthGandalf** [comment]
**Comment by ctrlaltca on 01 Jan 2009 18:56:24**
I've investigated a bit on this, and this is what i've found:
- Mirccryption is basically blowfish encryption + base64 , and kvirc plays fine with it (they share the codebase).
- Fish is not blowfish and is not base64; it's basically a modified blowfish using a uncommon keylength of 72 bytes (chars) instead of blowfish's 56 chars. The base64 encoder is not-standard, too.
- Mirccryption developers back in 2005 started accepting keylengths up to 80 chars in ecb mode.. probably breaking their blowfish "old" (ecb) mode.
- Using keylenghts of 72 chars offers a small amount of added security, but adds also the possibility of choosing a weak key that resulted in all 18 P sub−keys turning into 0 after
  being XORed with a 576 bit key (Short version: someone can trick you into using an empty password, and you won't notice). Not so clever.
- The real "boost" in security can be achieved using the cbc mode, instead of the default ecb one.

Short version of all the post: fish is not mircryption. Changing the ticket to a feature request: "implement fish support".

---

Summary: KVIrc mircryption support is incompatiable with mircryption! → Implement fish support
Type: Bug → Feature request
Version: 3.4.2 Shiny → 4.0 Insomnia

**Comment by DarthGandalf** [comment]
**Comment by ctrlaltca on 01 Jan 2009 19:00:35**
merged with #320: add DH1080 support (part of fish)

I'm struggling to understand why mircryption support is implemented, but not DH1080 initialisation. This is generally used to negotiate the mircryption keys to be used between 2 IRC users in a private conversation and is an incredibly useful way of negotiating a key to use securely.

**Comment by DarthGandalf** [comment]
**Modified by HelLViS69 on 19 Dec 2009 16:46:07**
Version: 4.0 Insomnia → 4.0rc2 Insomnia

**Comment by DarthGandalf** [comment]
**Modified by HelLViS69 on 02 Feb 2010 14:14:07**
Version: 4.0rc2 Insomnia → 4.0 Insomnia

**Comment by DarthGandalf** [comment]
**Modified by HelLViS69 on 27 Jun 2010 15:18:38**
Version: 4.0 Insomnia → 4.1 Equilibrium

**Comment by DarthGandalf** [comment]
**Comment by ambossarm on 31 Dec 2010 11:51:19**
A little push because that got asked a few times in IRC

A hint to bug #1011 that mircryption/fish is broken for libcryptousers

**Comment by DarthGandalf** [comment]
**Comment by ctrlaltca on 06 Feb 2011 23:40:16**
Initial fish support implemented in r5405 (dh1080 key exchange)<br/>
Longer keylengths won't be supported, have a look at r5429 to know why<br/>

---

Owner: pragmaware → ctrlaltca
Status: new → assigned

**Comment by DarthGandalf** [comment]
**Comment by ambossarm on 03 Mar 2012 22:36:48**
could we official ship that?
What do we need? Autoload module? Fancy buttons? DOCUMENTATION!

Had just a satisfied user in IRC, only said that some implementations would use ECB instead of CBC, but we should not implement ECB - never!

**Comment by DarthGandalf** [comment]
**Modified by HelLViS69 on 05 Jan 2015 16:03:04**
Resolution: none → fixed
Status: assigned → closed
